ompi/mca/osc/ucx/osc_ucx_comm.c

```diff
@@ -343,6 +334,11 @@ static inline int get_dynamic_win_info(uint64_t remote_addr, ompi_osc_ucx_module
     int insert = -1;
     int ret;
 
+    bool lock_needed = ompi_osc_need_acc_lock(module, ompi_comm_rank(module->comm));
```

Comment:
Shouldn't that be `ompi_osc_need_acc_lock(module, target);`? And same below for the lock call.

Reply:
Oh yes, good catch. It should be locking the target window. I will make the change.